app/assets/javascripts/lessons.js

```diff
@@ -1,121 +1,60 @@
-    }
-  });
-  $('.lc-end-wrapper').on('mouseleave', '.checkbox-container',function(e){
-    if(!$('.checkbox-container').hasClass('lc-disabled')){
-      $('.checkbox-container').removeClass('lc-hover');
-    }
-  });
-  
-  // Delegate an event listener to the completion wrapper for if the user clicks
-  // the "mark this lesson NOT completed" link
-  $('.lc-end-wrapper').on('click', '.lc-uncomplete-link', function(e){
-    e.preventDefault();
-
-    // Switch to AJAX waiting state, since it will be overwritten by the
-    // refresh from the unobtrusive javascript
-    $('.checkbox-container').addClass('lc-waiting');
-    
-    // submit the form
-    $(e.target).parents('form').submit();
-  });
-
-  /*
-   * Handlers for Octocat and the contributions links;
-   *    Clicking Octocat triggers a bootstrap popover with links to the lessons page in the Github repo,
-   *    and a link to the live site's Contributing page.
-   *    See also the lessons show.html.erb file and the partial _contribution_links.html.erb.
-  */
-  // Cache jquery reference to the cat, since its referenced everytime there is a click anywhere on the page.
-  var $octocat = $('.contribution-links');
-
-  // If the Octocat is clicked, put up bootstrap popover
-  $octocat.popover({
-    html: true,
-    title: 'Want to improve something in this lesson?'
-  });
-
-
-  // If popover is activated and user clicks somewhere other than popover itself or kitty, then hide popover.
-  //    Implementation Note: Popular solutions to catch off-element clicks to hide element that include call to
-  //    event.stopPropagation() will fail if that element outside Octokitty clicked-upon itself has stopPropagation!
-  //    The implementation below doesn't have that prob:
-  $(document).click(function(event) {
-    // If click is not on popover container or anything inside it, but click IS on ancestor of OctoKitty container
-    if (!($(event.target).is('.popover') ||  // (need these 2 cases cuz popover is also an ancestor of octokitty!)
-          $(event.target).parents().is('.popover') )  &&
-          $(event.target).parents().index($octocat) === -1 ) {
-      if ($('.popover').is(':visible'))       // only hide if popover is visible
-        $octocat.popover('hide');
-    }
-  });
+var markerClasses = [
+  'primary',
+  'coffee',
+  'whitish-grey',
+  'light-grey',
+  'greyish-brown'
+];
+
+function getElements(selector) {
+  return document.querySelectorAll(selector);
+}
+
+function kebabCase(text) {
+  return text.toLowerCase().match(/\w+/g).join('-');
+}
+
+function setTargetForExternalLinks() {
+  getElements('.lesson-content a[href^=http]').forEach(function (externalLink) {
+    externalLink.setAttribute('target', '_blank');
+  });
+}
+
+function navigationElement(markerClass, headingText) {
+  return (
+    '<div class="lesson-navigation__element">' +
+    '<div class="lesson-navigation__marker lesson-navigation__marker--' + markerClass + '"></div>' +
```

Comment:
` + markerClass +`
Just curious, why do we need to append this instead of having it in within the string?